### DISCUSSION TITLE
How to return the updated entity from an update?

### DISCUSSION THREAD

**Discussion by matthewkooshad** [simolus3/drift#2494] [Q&A] [answered]
i took another look at: https://drift.simonbinder.eu/docs/getting-started/writing_queries/#updates-and-deletes
as i'm wanting to see if any ideas were shared about how i could get the entity after an update.

i like the cool `copyWith` method, which the result of that is what i want.
but, if i use `UpdateCompanion` or `UpdateStatement`, to update a row where i don't have the entity at hand to use `copyWith`, how can i similarly get the updated entity returned after the update?
(supposing i have a Stuffs table and use update(stuffs) or StuffsCompanion in doing the update, similar to how `into(stuffs).insertReturning` conveniently does)
thanks!

**Comment by simolus3** [comment, marked as answer]
Have you seen [`UpdateStatement.writeReturning`](https://drift.simonbinder.eu/api/drift/updatestatement/writereturning)? It creates a returning clause and maps results for update statements. It returns a list since an update can affect more than one row, but if you know you're only updating a single entity, you can just use single on it:

```dart
final stmt = update(stuffs)..whereSamePrimaryKey(something);
final result = await stmt.writeReturning(yourUpdateCompanion);
return result.single;
```

**Reply by matthewkooshad** [reply]
I wasn't aware of that, thanks! I'll give it a shot and report here if I face an issue with it. Thanks for the quick help!
